lib/scripting/post-process/assign.ts

```diff
@@ -17,26 +17,36 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
-import { Expression, ExpressionStatement, MemberExpression } from 'estree';
-import { Token } from 'moo';
-import * as estree from '../estree';
+import { assignmentExpression, expressionStatement } from '../estree';
+import { ESIToken } from '../grammar/grammar';
 
-export function stmt1(result: [MemberExpression, null, Token, null, Expression]): ExpressionStatement {
-	const left = result[0];
-	const right = result[4];
-	return estree.expressionStatement(estree.assignmentExpression(left, '=', right));
+export function ppAssign(node: ESIToken): any {
+	switch (node.type) {
+		case 'RegularAssignmentStatement':
+		case 'RegularAssignmentStatementInline':
+			return ppRegularAssignmentStatement(node);
+		case 'SetAssignmentStatement':
+		case 'SetAssignmentStatementInline':
+			return ppSetAssignmentStatement(node);
+	}
+	return null;
```

Comment:
is this an error of no type is found?

Reply:
The post processors are more like handlers, we call all them one by one until one finally returns an estree.

Reply:
The post processors are more like handlers, we call all them one by one until one finally returns an estree.